packages/mongodb-memory-server-core/src/util/MongoBinary.ts

```diff
@@ -147,30 +164,40 @@ export default class MongoBinary {
           .split('\n')[0]
           .split(' ')[2];
 
-        if (version !== LATEST_VERSION && version !== binaryVersion) {
+        if (options.version !== LATEST_VERSION && options.version !== binaryVersion) {
           // we will log the version number of the system binary and the version requested so the user can see the difference
           log(
             'MongoMemoryServer: Possible version conflict\n' +
               `  SystemBinary version: ${binaryVersion}\n` +
-              `  Requested version:    ${version}\n\n` +
+              `  Requested version:    ${options.version}\n\n` +
               '  Using SystemBinary!'
           );
         }
       }
     }
 
     if (!binaryPath) {
-      binaryPath = await this.getCachePath(version);
+      binaryPath = this.getCachePath(options.version);
     }
 
     if (!binaryPath) {
       binaryPath = await this.getDownloadPath(options);
     }
 
-    log(`MongoBinary: Mongod binary path: ${binaryPath}`);
+    if (!binaryPath) {
+      throw new Error(
```

Comment:
added an Error for more security in returns (because all of the functions above can return an false-y value

Reply:
👍